.github/workflows/external-integration.yml

```diff
@@ -0,0 +1,84 @@
+name: External Integration
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled]
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  azure-rest-api-specs:
+    name: Azure REST API Specs
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'int:azure-specs') || github.event_name == 'workflow_dispatch'
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+
+      - name: Checkout Azure REST API specs
+        uses: actions/checkout@v4
+        with:
+          repository: Azure/azure-rest-api-specs
+          path: azure-rest-api-specs
+          ref: typespec-next
```

Comment:
@timotheeguerin Should we be merging `origin/main` into `typespec-next` like the specs repo does?

Reply:
talked offline will keep it as it is for now and update later if its confusing